app/updatemanager.go

```diff
@@ -200,6 +200,25 @@ func (c *ControlMapPool) insertMap(keepPredicate func(*updatecontrolmap.UpdateCo
 	c.announceUpdate()
 }
 
+// HasServerControlMap - Returns true in the event that a control map matches
+// the given deploymentID.
+func (c *ControlMapPool) HasServerControlMap(deploymentID string) bool {
```

Comment:
From an isolated perspective, there is nothing server-specific about this. I would just call it `HasControlMap`, and then call the argument `id`. The fact that it's used to fetch a server map is something that only the caller cares about.

This is, strictly speaking very similar to calling `Get` and see if the `len(result) > 0`, but I will not object to a convenience function. 🙂

Reply:
Ohhh, true!

I like the `Get` part!

Rewriting, and just adding a comment :raised_hands: 

Thanks!

Reply:
Ended up using `Get` in the `HasServerControlMap` function.

I left the name as `HasServerControlMap`.

But let's have a small discussion on:

I chose the `Server` prefix, just to make it clear to the caller what it is used for, even though the function itself can be used more flexibly.

But all open to changing it.

Floor is yours

Reply:
I would vote for changing the name. Naming functions based on the intention of the caller is not generally good design, it's better to isolate it to what the function does. It's better to state the intended use with a comment where you use it.

Reply:
As a practical example, consider if this function was in a different Go package, then it would be much clearer that the function should not know anything about the caller.

Reply:
You got me :)

Reply:
Gamle ørn :eagle:

Reply:
😂